source/MaterialXView/Viewer.cpp

```diff
@@ -1088,11 +1094,145 @@ void Viewer::drawContents()
     }
 }
 
+mx::MeshStreamPtr Viewer::createUvPositionStream(mx::MeshPtr mesh, 
+                                                 const std::string& uvStreamName, 
+                                                 unsigned int index,
+                                                 const std::string& positionStreamName)
+{
+    mx::MeshStreamPtr uvStream3D = mesh->getStream(positionStreamName);
+    if (!uvStream3D)
+    {
+        uvStream3D = mx::MeshStream::create(positionStreamName, mx::MeshStream::POSITION_ATTRIBUTE, 0);
+        mesh->addStream(uvStream3D);
+
+        mx::MeshStreamPtr uvStream2D = mesh->getStream(uvStreamName, index);
+        mx::MeshFloatBuffer &uvPos2D = uvStream2D->getData();
+        mx::MeshFloatBuffer &uvPos3D = uvStream3D->getData();
+        size_t uvCount = uvPos2D.size() / 2;
+        uvPos3D.resize(uvCount * 3);
+        const float MAX_FLOAT = std::numeric_limits<float>::max();
+        mx::Vector3 boxMin = { MAX_FLOAT, MAX_FLOAT, 0.0f };
+        mx::Vector3 boxMax = { -MAX_FLOAT, -MAX_FLOAT, 0.0f };
+        for (size_t i = 0; i < uvCount; i++)
+        {
+            float u = uvPos2D[i * 2];
+            uvPos3D[i * 3] = u;
+            float v = uvPos2D[i * 2 + 1];
+            uvPos3D[i * 3 + 1] = v;
+            uvPos3D[i * 3 + 2] = 0.0f;
+
+            boxMin[0] = std::min(u, boxMin[0]);
+            boxMin[1] = std::min(v, boxMin[1]);
+            boxMax[0] = std::max(u, boxMax[0]);
+            boxMax[1] = std::max(v, boxMax[1]);
+        }
+
+        mx::Vector3 sphereCenter = (boxMax + boxMin) / 2.0;
+        float sphereRadius = (sphereCenter - boxMin).getMagnitude();
+        _uvScale[0] = 2.0f / sphereRadius;
+        _uvScale[1] = 2.0f / sphereRadius;
+        _uvScale[2] = 1.0f;
+        _uvTranslation[0] = -sphereCenter[0];
+        _uvTranslation[1] = -sphereCenter[1];
+        _uvTranslation[2] = 0.0f;
+    }
+
+    return uvStream3D;
+}
+
+void Viewer::drawScene2D()
+{
+    // Create uv shader if it does not exist
+    if (!_wireMaterialUV)
+    {
+        const std::string shaderName("__UV_WIRE_SHADER_NAME__");
+        const mx::Color3 color(1.0f);
+        _wireMaterialUV = Material::create();
+        try
+        {
+            _wireMaterialUV->generateConstantShader(_genContext, _stdLib, shaderName, color);
+        }
+        catch (std::exception& e)
+        {
+            _wireMaterialUV = nullptr;
+            std::cerr << "Failed to generate uv wire shader: " << e.what();
+            return;
+        }
+    }
+
+    GLShaderPtr shader = _wireMaterialUV ? _wireMaterialUV->getShader() : nullptr;
```

Comment:
Null check on _wireMaterialUV is already done above.

Reply:
Will fix.